Arrowgene.Ddon.GameServer/Handler/AreaGetAreaSupplyHandler.cs

```diff
@@ -0,0 +1,63 @@
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Logging;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class AreaGetAreaSupplyHandler : GameRequestPacketHandler<C2SAreaGetAreaSupplyReq, S2CAreaGetAreaSupplyRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(AreaGetAreaSupplyHandler));
+
+        public AreaGetAreaSupplyHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CAreaGetAreaSupplyRes Handle(GameClient client, C2SAreaGetAreaSupplyReq request)
+        {
+            if (!client.Character.AreaSupply.ContainsKey(request.AreaId))
+            {
+                throw new ResponseErrorException(ErrorCode.ERROR_CODE_AREAMASTER_SUPPLY_NOT_AVAILABLE);
+            }
+            List<CDataRewardItemInfo> supply = client.Character.AreaSupply[request.AreaId];
+
+            // I don't know what this "StorageType" is actually corresponding to, but there are only two values permitted
+            // by the Area Supply UI.
+            bool toBag = request.StorageType switch
+            {
+                19 => true,
+                20 => false,
+                _ => throw new ResponseErrorException(ErrorCode.ERROR_CODE_AREAMASTER_INVALID_SUPPLY_STORAGE),
```

Comment:
We've seen this 19/20 values meaning to bag/to storage a lot, maybe it should be time to move them into an enum? 🤔

Reply:
I agree, but do we know what it really means?

Reply:
Besides 19 meaning bag and 20 being storage, not really, maybe the mobile app has some answers